lib/modules/datasource/bitbucket-server-tags/index.ts

```diff
@@ -0,0 +1,194 @@
+  BitbucketServerTag,
+  BitbucketServerTags,
+} from './schema';
+
+export class BitbucketServerTagsDatasource extends Datasource {
+  static readonly id = 'bitbucket-server-tags';
+
+  override http = new BitbucketServerHttp(BitbucketServerTagsDatasource.id);
+
+  static readonly defaultRegistryUrls = ['https://bitbucket.org'];
+
+  static readonly sourceUrlSupport = 'package';
+  static readonly sourceUrlNote =
+    'The source URL is determined by using the `packageName` and `registryUrl`.';
+
+  static readonly cacheNamespace: PackageCacheNamespace = `datasource-${BitbucketServerTagsDatasource.id}`;
+
+  constructor() {
+    super(BitbucketServerTagsDatasource.id);
+  }
+
+  static getRegistryURL(registryUrl?: string): string {
+    return (
+      registryUrl?.replace(regEx(/\/rest\/api\/1.0$/), '') ??
+      this.defaultRegistryUrls[0]
+    );
+  }
+
+  static getSourceUrl(
+    projectKey: string,
+    repositorySlug: string,
+    registryUrl?: string,
+  ): string {
+    const url = BitbucketServerTagsDatasource.getRegistryURL(registryUrl);
+    return `${ensureTrailingSlash(url)}projects/${projectKey}/repos/${repositorySlug}`;
+  }
+
+  static getApiUrl(registryUrl?: string): string {
+    const res = BitbucketServerTagsDatasource.getRegistryURL(registryUrl);
+    return `${ensureTrailingSlash(res)}rest/api/1.0/`;
+  }
+
+  static getCacheKey(
+    registryUrl: string | undefined,
+    repo: string,
+    type: string,
+  ): string {
+    return `${BitbucketServerTagsDatasource.getRegistryURL(registryUrl)}:${repo}:${type}`;
+  }
+
+  // getReleases fetches list of tags for the repository
+  @cache({
+    namespace: BitbucketServerTagsDatasource.cacheNamespace,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      BitbucketServerTagsDatasource.getCacheKey(
+        registryUrl,
+        packageName,
+        'tags',
+      ),
+  })
+  async getReleases(config: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const { registryUrl, packageName } = config;
+    const [projectKey, repositorySlug] = packageName.split('/');
+
+    const result = Result.parse(config, ReleasesConfig)
+      .transform(({ registryUrl }) => {
+        const url = `${BitbucketServerTagsDatasource.getApiUrl(registryUrl)}projects/${projectKey}/repos/${repositorySlug}/tags`;
+
+        return this.http.getJsonSafe(
+          url,
+          { paginate: true },
+          BitbucketServerTags,
+        );
+      })
+      .transform((tags) =>
+        tags.map(({ displayId, hash }) => ({
+          version: displayId,
+          gitRef: displayId,
+          newDigest: hash ?? undefined,
+        })),
+      )
+      .transform((versions): ReleaseResult => {
+        return {
+          sourceUrl: BitbucketServerTagsDatasource.getSourceUrl(
+            projectKey,
+            repositorySlug,
+            registryUrl,
+          ),
+          registryUrl:
+            BitbucketServerTagsDatasource.getRegistryURL(registryUrl),
```

Comment:
why do we need to compute the registry url again? can't we reuse `registryUrl` ?

Reply:
Depending on how the datasource is used (directly by users or during changelog retrieval) `registryUrl` could be `bitbucket.server.org/rest/api/1.0` or `bitbucket.server.org`. 

Calling `getRegistryURL()` here ensures that the value returned for releases is always without `/rest/api/1.0` suffix, which is how this value also looks like in other datasources.